sunpy/net/helio/hec.py

```diff
@@ -78,13 +90,90 @@ def __init__(self, link=None):
         transport = Transport(session=session)
         self.hec_client = Client(link, transport=transport)
 
-    def time_query(self, start_time, end_time, table=None, max_records=None):
+    @classmethod
+    def _can_handle_query(cls, *query):
+        required = {a.Time}
+        optional = {ha.MaxRecords, ha.TableName}
+        return cls.check_attr_types_in_query(query, required, optional)
+
+    @classmethod
+    def _attrs_module(cls):
+        return 'helio', 'sunpy.net.helio.attrs'
+
+    def search(self, *args, **kwargs):
         """
         The simple interface to query the wsdl service.
 
         Used to utilize the service's TimeQuery() method, this is a simple
         interface between the sunpy module library and the web-service's API.
 
+        Examples
+        --------
+        >>> from sunpy.net.helio import attrs as ha
+        >>> from sunpy.net import attrs as a, Fido
+        >>> timerange = a.Time('2005/01/03', '2005/12/03')
+        >>> res = Fido.search(timerange, ha.MaxRecords(10),
+        ...                   ha.TableName('rhessi_hxr_flare'))  # doctest: +REMOTE_DATA
+        >>> res  #doctest: +REMOTE_DATA
+        <sunpy.net.fido_factory.UnifiedResponse object at ...>
+        Results from 1 Provider:
+        <BLANKLINE>
+        10 Results from the HECClient:
+        hec_id      time_start          time_peak      ... energy_kev flare_number
+        ------ ------------------- ------------------- ... ---------- ------------
+         31463 2005-01-03T01:37:36 2005-01-03T01:37:54 ...          6      5010320
+         31464 2005-01-03T01:51:36 2005-01-03T01:59:18 ...         12      5010301
+         31465 2005-01-03T03:26:28 2005-01-03T03:42:50 ...          6      5010332
+         31466 2005-01-03T03:46:04 2005-01-03T04:07:10 ...         12      5010302
+         31467 2005-01-03T05:00:24 2005-01-03T05:00:30 ...          6      5010313
+         31468 2005-01-03T06:40:48 2005-01-03T06:42:46 ...          6      5010314
+         31469 2005-01-03T08:27:56 2005-01-03T08:28:26 ...          6      5010334
+         31470 2005-01-03T09:31:00 2005-01-03T09:33:34 ...          6      5010322
+         31471 2005-01-03T09:34:52 2005-01-03T09:59:46 ...          6      5010336
+         31472 2005-01-03T11:06:48 2005-01-03T11:07:18 ...         12      5010304
+        <BLANKLINE>
+        <BLANKLINE>
+        """
+        qrdict = {}
+        for elem in args:
+            if isinstance(elem, a.Time):
+                qrdict['Time'] = elem
+            elif isinstance(elem, ha.MaxRecords):
+                qrdict['max_records'] = elem.value
+            elif isinstance(elem, ha.TableName):
+                qrdict['table_name'] = elem.value
+            else:
+                raise ValueError(
+                    "HECClient can not add {} to the map_ dictionary to pass "
+                    "to the Client.".format(elem.__class__.__name__))
```

Comment:
This error message is too technical. What we are really asserting here is a subset of allowed attr classes?

Reply:
Looks like it?

Reply:
Fixed.